dynawo/sources/Models/CPP/Common/DYNModelCPPImpl.h

```diff
@@ -254,7 +254,7 @@ class ModelCPP::Impl : public ModelCPP {
   /**
    * @copydoc ModelCPP::checkDataCoherence()
    */
-  void checkDataCoherence(const double& t);
+  virtual void checkDataCoherence(const double& t) = 0;
```

Comment:
Not needed as already in DYNModelCPP.h

Reply:
I don't know what is the good practice but in the current state, all the methods are redeclared in ModelCPP::Impl, even if they are just completely identical to the declaration in DYNModelCPP.h (see evalF, evalJt, evalG, etc.) so I can do it but then the code is no longer "uniformized", what I never really like.
What do you think?

Reply:
I don't know what is the best practice but all declarations are currenly duplicated in DYNModelCPPImpl.h (in addition to DYNModel.h). If I delete this one, we will lose somehow the implementation uniformity, what I never really like. What do you think?

Reply:
good point I didn't see that all purely virtuals methods were declared in the implementation. So yes let's keep this homogeneous and keep this declaration

Reply:
why use pure virtual and not just virtual ? Other cpp models now have to define an empty methods that can just be overriden in the model that needs it (OmegaRef).

Reply:
You are right. And it is already the case for a few other methods: initSubBuffers, setFequationsInit at least ; so it could make sense to do it this way. 
The positive point I see with the obligation to define the empty method is that it forces to a certain extent the developer to think about this method and its use in its model.

Anyway, I think we can modify to virtual. As we still have an open subject on the error type to use (do we need a specific error type for frequency problems), we can do both at the same time. I will reopen the issue.